warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -982,12 +982,13 @@ public T next() {
             final Iterator<Tuple3<Key,Document,Map<String,Object>>> itrWithContext;
             if (this.isTermFrequenciesRequired()) {
                 
-                // The TFFunction can only prune non index-only fields
-                Set<String> tfIndexOnlyFields = Sets.intersection(getTermFrequencyFields(), getIndexOnlyFields());
+                // First source fetches term offsets
+                SortedKeyValueIterator<Key,Value> firstSource = sourceDeepCopy.deepCopy(myEnvironment);
+                // Second source fetches tf field value pairs from the field index for delayed (negated) phrases
+                SortedKeyValueIterator<Key,Value> secondSource = sourceDeepCopy.deepCopy(myEnvironment);
                 
-                Function<Tuple2<Key,Document>,Tuple3<Key,Document,Map<String,Object>>> tfFunction;
-                tfFunction = TFFactory.getFunction(getScript(documentSource), getContentExpansionFields(), getTermFrequencyFields(), this.getTypeMetadata(),
-                                super.equality, getEvaluationFilter(), sourceDeepCopy.deepCopy(myEnvironment), tfIndexOnlyFields);
+                Function<Tuple2<Key,Document>,Tuple3<Key,Document,Map<String,Object>>> tfFunction = buildTfFunction(getScript(documentSource), firstSource,
+                                secondSource);
```

Comment:
Perhaps we allow the function to `deepCopy` the iterators if it needs to, not all function implementations would need two sources?

Reply:
Delaying the second deep copy makes sense, especially given that it should only happen if a query contains negations.